packages/babel-helper-validator-option/src/validator.js

```diff
@@ -0,0 +1,77 @@
+// @flow
+import { findSuggestion } from "./find-suggestion.js";
+
+export class OptionValidator {
+  declare descriptor: string;
+  constructor(descriptor: string) {
+    this.descriptor = descriptor;
+  }
+
+  /**
+   * Validate if the given `options` follow the name of keys defined in the `TopLevelOptionShape`
+   *
+   * @param {Object} options
+   * @param {Object} TopLevelOptionShape
+   *   An object with all the valid key names that `options` should be allowed to have
+   *   The property values of `TopLevelOptionShape` can be arbitrary
+   * @memberof OptionValidator
+   */
+  validateTopLevelOptions(options: Object, TopLevelOptionShape: Object): void {
+    const validOptionNames = Object.keys(TopLevelOptionShape);
+    for (const option of Object.keys(options)) {
+      if (!validOptionNames.includes(option)) {
+        throw new Error(
+          `${this.descriptor}: '${option}' is not a valid top-level option.
+- Maybe you meant to use '${findSuggestion(option, validOptionNames)}'?`,
+        );
+      }
+    }
+  }
+
+  // note: we do not consider rewrite them to high order functions
```

Comment:
Probably don't need this comment, but if we want to make it clear that this should be refactored in the future maybe we can add a `TODO`?

Reply:
In fact it is quite opposite: It is to leave it as-is (for less call stacks when error is thrown) _unless_ we are going to support number-only preset options in the future, which is quite unlikely.